server/src/main/java/org/opensearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -91,9 +92,17 @@ public class IndexNameExpressionResolver {
     private final List<ExpressionResolver> expressionResolvers = List.of(dateMathExpressionResolver, wildcardExpressionResolver);
 
     private final ThreadContext threadContext;
+    private final String[] systemIndices;
 
     public IndexNameExpressionResolver(ThreadContext threadContext) {
         this.threadContext = Objects.requireNonNull(threadContext, "Thread Context must not be null");
+        this.systemIndices = new String[0];
+    }
+
+    public IndexNameExpressionResolver(ThreadContext threadContext, SystemIndices systemIndices) {
```

Comment:
You should be able to put `@InternalApi` on this constructor since plugins should not be instantiating instances of this class.

Reply:
Added